x-pack/legacy/plugins/reporting/export_types/png/server/lib/generate_png.ts

```diff
@@ -5,65 +5,46 @@
  */
 
 import * as Rx from 'rxjs';
-import { toArray, mergeMap } from 'rxjs/operators';
+import { map } from 'rxjs/operators';
 import { LevelLogger } from '../../../../server/lib';
 import { ServerFacade, ConditionalHeaders } from '../../../../types';
 import { screenshotsObservableFactory } from '../../../common/lib/screenshots';
 import { PreserveLayout } from '../../../common/layouts/preserve_layout';
 import { LayoutParams } from '../../../common/layouts/layout';
 
-interface ScreenshotData {
-  base64EncodedData: string;
-}
-
-interface UrlScreenshot {
-  screenshots: ScreenshotData[];
-}
-
 export function generatePngObservableFactory(server: ServerFacade) {
   const screenshotsObservable = screenshotsObservableFactory(server);
-  const captureConcurrency = 1;
-
-  // prettier-ignore
-  const createPngWithScreenshots = async ({ urlScreenshots }: { urlScreenshots: UrlScreenshot[] }): Promise<string> => {
-    if (urlScreenshots.length !== 1) {
-      throw new Error(
-        `Expected there to be 1 URL screenshot, but there are ${urlScreenshots.length}`
-      );
-    }
-    if (urlScreenshots[0].screenshots.length !== 1) {
-      throw new Error(
-        `Expected there to be 1 screenshot, but there are ${urlScreenshots[0].screenshots.length}`
-      );
-    }
-
-    return urlScreenshots[0].screenshots[0].base64EncodedData;
-  };
 
   return function generatePngObservable(
     logger: LevelLogger,
     url: string,
     browserTimezone: string,
     conditionalHeaders: ConditionalHeaders,
     layoutParams: LayoutParams
-  ): Rx.Observable<string> {
+  ): Rx.Observable<Buffer> {
     if (!layoutParams || !layoutParams.dimensions) {
       throw new Error(`LayoutParams.Dimensions is undefined.`);
     }
 
     const layout = new PreserveLayout(layoutParams.dimensions);
-    const screenshots$ = Rx.of(url).pipe(
-      mergeMap(
-        iUrl =>
-          screenshotsObservable({ logger, url: iUrl, conditionalHeaders, layout, browserTimezone }),
-        (jUrl: string, screenshot: UrlScreenshot) => screenshot,
-        captureConcurrency
-      )
+    const screenshots$ = screenshotsObservable({
+      logger,
+      url,
+      conditionalHeaders,
+      layout,
+      browserTimezone,
+    }).pipe(
+      map(urlScreenshots => {
+        if (urlScreenshots.screenshots.length !== 1) {
+          throw new Error(
+            `Expected there to be 1 screenshot, but there are ${urlScreenshots.screenshots.length}`
+          );
+        }
+
+        return urlScreenshots.screenshots[0].base64EncodedData;
+      })
     );
 
-    return screenshots$.pipe(
-      toArray(),
```

Comment:
I changed this pipeline so that it doesn't unnecessarily wrap the generated data in an array, and then unwrap it later.